test/integration/data-types.test.js

```diff
@@ -273,4 +273,29 @@ describe(Support.getTestDialectTeaser('DataTypes'), function() {
      // No dialects actually allow us to identify that we get an enum back..
     testFailure(Type);
   });
+
+  it('calls parse for GEOMETRY with null buffer', function () {
+   var Type = new Sequelize.GEOMETRY();
+
+   if (current.dialect.supports.GEOMETRY) {
+     current.refreshTypes();
+
+     var User = current.define('user', {
+       field: Type
+     }, {
+       timestamps: false
+     });
+
+     //This case throw unhandled exception
+     return current.sync({ force: true }).then(function () {
+       return User.create({
+          field: { type: "Point", coordinates: [] }
+        });
+      }).then(function () {
+        return User.findAll();
+      }).then(function(users){
+        expect(users[0]).to.be.not.null;
+      });
+   }
+ });
```

Comment:
This test needs to be expanded a bit.
Change the description to `should parse an empty GEOMETRY field`.

Add a comment at `field: { type: "Point", coordinates: [] }` about this generating a `NULL` insert for MySQL.
Add a verification on the fetched user that values are as expected.